tools/jnimarshalmethod-gen/App.cs

```diff
@@ -245,7 +245,16 @@ void CreateMarshalMethodAssembly (string path)
 
 			PrepareTypeMap (ad.MainModule);
 
-			foreach (var type in assembly.DefinedTypes) {
+			IEnumerable<TypeInfo> types = null;
+			try {
+				types = assembly.DefinedTypes;
+			} catch (ReflectionTypeLoadException e) {
+				types = (IEnumerable<TypeInfo>) e.Types;
```

Comment:
what is the difference between `Types` and `DefinedTypes`? Is it ok to just use `Types` here?

Reply:
It is. the `e.Types` are the successfully  loaded defined types. I noticed one problem though, when rereading the documentation. There are null types in the results in case of load exceptions, so I added the necessary check to the following cycle.